### DISCUSSION TITLE
Softconsole support for Square root function

### DISCUSSION THREAD

**Discussion by josh25219** [polarfire-soc/polarfire-soc-documentation#207] [How to] [answered]
The release notes for Softconsole 2021.1 indicate lacking support for floating point operations such as sqrt.  
_Note the newlib doesn’t have fully ported floating point support for RISC-V and is not utilizing some floating point
instructions even if they are supported by the hardware (fmin/fmax/fsqrt…) and some functions might accidentally
include soft double support. This can lead to the application footprint growing unexpectedly. If really needed, it’s
possible to avoid these newlib functions and replace them with own implementations using inline assembly._

It isn't mentioned in any of the later release note versions.   Does the latest Softconsole version 2022.2 support sqrt functions?

Thanks
Josh

**Comment by AntonKrug** [comment, marked as answer]
Hi Josh, look at the `common.cpp` and `common.hpp` inside the `miv-rv32imaf-cpp` bundled project.
